client/v3/client_test.go

```diff
@@ -244,10 +244,52 @@ func TestAuthTokenBundleNoOverwrite(t *testing.T) {
 	}
 }
 
+func TestSyncFiltersMembers(t *testing.T) {
+	c, _ := NewClient(t, Config{Endpoints: []string{"http://254.0.0.1:12345"}})
+	defer c.Close()
+	c.Cluster = &mockCluster{}
+	c.Sync(context.Background())
+
+	endpoints := c.Endpoints()
+	if len(endpoints) != 1 || endpoints[0] != "http://254.0.0.3:12345" {
+		t.Error("Client.Sync uses learner and/or non-started member client URLs")
+	}
+}
+
 type mockAuthServer struct {
 	*etcdserverpb.UnimplementedAuthServer
 }
 
 func (mockAuthServer) Authenticate(context.Context, *etcdserverpb.AuthenticateRequest) (*etcdserverpb.AuthenticateResponse, error) {
 	return &etcdserverpb.AuthenticateResponse{Token: "mock-token"}, nil
 }
+
+type mockCluster struct{}
```

Comment:
Suggest to add a field something like  `members []*etcdserverpb.Member` into `mockCluster`, and update MemberList to return this `members` field instead of a hard code list.

Reply:
Done, good suggestion, thanks